### PR TITLE
Add pemutil.BundleCertificate

### PR DESCRIPTION
**Pull request by areed**
#### Name of feature:
AppendToBundle

#### Pain or issue this feature alleviates:
Easily add a certificate to a bundle if it's not already in the bundle.

#### Why is this important to the project (if not answered above):
Supports use cases where you don't want to needlessly append a certificate to a PEM-encoded certificate bundle if it's already in the bundle.

#### Supporting links/other PRs/issues:

💔Thank you!

### CODE REVIEW

File: pemutil/pem.go [hunk 1]
```diff
@@ -707,3 +708,30 @@ func ParseSSH(b []byte) (interface{}, error) {
 		return nil, errors.Errorf("unsupported key type %T", key)
 	}
 }
+
+// AppendToBundle adds a PEM-encoded certificate to a PEM-encoded certificate
+// bundle if not already in the bundle.
+func AppendToBundle(bundlePEM, certPEM []byte) ([]byte, bool, error) {
```
Comment (maraino) [inline pemutil/pem.go:714]: A variadic method might be better:
```go
func AppendToBundle(bundlePEM []byte, certsPEM ...[]byte) ([]byte, bool, error)
```

File: pemutil/pem.go [hunk 2]
```diff
@@ -707,3 +708,30 @@ func ParseSSH(b []byte) (interface{}, error) {
...
+	bundle, err := ParseCertificateBundle(bundlePEM)
+	if err != nil {
+		return nil, false, fmt.Errorf("invalid bundle: %w", err)
+	}
+	cert, err := ParseCertificate(certPEM)
+	if err != nil {
+		return nil, false, fmt.Errorf("invalid cert: %w", err)
+	}
+
+	sums := make(map[[28]byte]bool, len(bundle))
```
Comment (maraino) [inline pemutil/pem.go:724]:
```suggestion
	sums := make(map[[sha256.Size224]byte]bool, len(bundle))
```

**Review by maraino — CHANGES_REQUESTED** [review]

File: pemutil/pem.go (hunk 1 above)
Comment (maraino) [reply pemutil/pem.go:714]: And perhaps rename it to `BundleCertificate`, something related to certificates (better than `AppendtoCertificateBundle`. The pemutil package manages keys too.

Comment (areed) [reply pemutil/pem.go]: a084886

File: pemutil/pem.go (hunk 2 above)
Comment (areed) [reply pemutil/pem.go]: a084886

File: pemutil/pem.go [hunk 3]
```diff
@@ -707,3 +708,35 @@ func ParseSSH(b []byte) (interface{}, error) {
...
+	for i := range bundle {
+		sums[sha256.Sum224(bundle[i].Raw)] = true
+	}
+
+	modified := false
+
+	for i := range certsPEM {
+		cert, err := ParseCertificate(certsPEM[i])
+		if err != nil {
+			return nil, false, fmt.Errorf("invalid cert %d: %w", i, err)
```
Comment (maraino) [inline pemutil/pem.go:730]:
```suggestion
			return nil, false, fmt.Errorf("invalid certificate %d: %w", i, err)
```

File: pemutil/pem_test.go [hunk 4]
```diff
@@ -1177,3 +1177,51 @@ func TestReadCertificateRequest(t *testing.T) {
...
+		bundle string
+		certs  []string
+		added  bool
+		err    error
+	}{
+		{"append", "testdata/bundle.crt", []string{"testdata/ca.crt"}, true, nil},
+		{"two", "testdata/bundle.crt", []string{"testdata/ca.crt", "testdata/ca2.crt"}, true, nil},
+		{"none", "testdata/bundle.crt", nil, false, nil},
+		{"found", "testdata/ca.crt", []string{"testdata/ca.crt"}, false, nil},
+		{"bad cert", "testdata/bundle.crt", []string{"testdata/badca.crt"}, false, errors.New("invalid cert 0")},
```
Comment (maraino) [inline pemutil/pem_test.go:1193]:
```suggestion
		{"bad cert", "testdata/bundle.crt", []string{"testdata/badca.crt"}, false, errors.New("invalid certificate 0")},
```

**Review by maraino — APPROVED** [review]

**Pull request merged**